File: Examples/c/04_conditionals.c
/*
 * Conditional Statements
 * 
 * Demonstrates:
 * - if, else if, else statements
 * - Comparison operators
 * - Logical operators
 * - switch statement
 */

#include <stdio.h>

int main() {
    int score;
    char grade;
    
    printf("=== Grade Calculator ===\n\n");
    printf("Enter your score (0-100): ");
    scanf("%d", &score);
    
    // Validate input
    if (score < 0 || score > 100) {
        printf("Error: Score must be between 0 and 100!\n");
        return 1;
    }
    
    // Determine letter grade
    if (score >= 90) {
        grade = 'A';
        printf("Excellent! Grade: %c\n", grade);
    } else if (score >= 80) {
        grade = 'B';
        printf("Good job! Grade: %c\n", grade);
    } else if (score >= 70) {
        grade = 'C';
        printf("Average. Grade: %c\n", grade);
    } else if (score >= 60) {
        grade = 'D';
        printf("Below average. Grade: %c\n", grade);
    } else {
        grade = 'F';
        printf("Failing. Grade: %c\n", grade);
    }
    
    // Switch statement example
    printf("\nFeedback: ");
    switch (grade) {
        case 'A':
            printf("Outstanding work!\n");
            break;
        case 'B':
            printf("Keep it up!\n");
            break;
        case 'C':
            printf("Room for improvement.\n");
            break;
        case 'D':
            printf("Need more study time.\n");
            break;
        case 'F':
            printf("Please see instructor.\n");
            break;
        default:
            printf("Unknown grade.\n");
    }
    
    return 0;
}

File: Examples/pascal/04_conditionals.pas
{
  Conditional Statements
  
  Demonstrates:
  - If-Then-Else statements
  - Case statement (Pascal's switch)
  - Boolean expressions
  - Nested conditionals
}

Program Conditionals;
Var
  score: Integer;
  grade: Char;

Begin
  WriteLn('=== Grade Calculator ===');
  WriteLn;
  
  Write('Enter your score (0-100): ');
  ReadLn(score);
  
  { Validate input }
  If (score < 0) Or (score > 100) Then
  Begin
    WriteLn('Error: Score must be between 0 and 100!');
    Halt(1);
  End;
  
  { Determine letter grade }
  If score >= 90 Then
  Begin
    grade := 'A';
    WriteLn('Excellent! Grade: ', grade);
  End
  Else If score >= 80 Then
  Begin
    grade := 'B';
    WriteLn('Good job! Grade: ', grade);
  End
  Else If score >= 70 Then
  Begin
    grade := 'C';
    WriteLn('Average. Grade: ', grade);
  End
  Else If score >= 60 Then
  Begin
    grade := 'D';
    WriteLn('Below average. Grade: ', grade);
  End
  Else
  Begin
    grade := 'F';
    WriteLn('Failing. Grade: ', grade);
  End;
  
  { Case statement example }
  WriteLn;
  Write('Feedback: ');
  Case grade Of
    'A': WriteLn('Outstanding work!');
    'B': WriteLn('Keep it up!');
    'C': WriteLn('Room for improvement.');
    'D': WriteLn('Need more study time.');
    'F': WriteLn('Please see instructor.');
  Else
    WriteLn('Unknown grade.');
  End;
End.
